sphinx/util/inspect.py

```diff
@@ -317,15 +378,14 @@ def iswrappedcoroutine(obj: Any) -> bool:
     return inspect.iscoroutinefunction(obj)
 
 
-def isproperty(obj: Any) -> bool:
-    """Check if the object is property."""
+def isproperty(obj: Any) -> TypeGuard[property | cached_property]:
+    """Check if the object is property (possibly cached)."""
     return isinstance(obj, (property, cached_property))
 
 
 def isgenericalias(obj: Any) -> bool:
```

Comment:
In case you're wondering, I won't TypeGuard that one because the typing private interface is too fragile between versions.